Modules/Microsoft365DSC/DSCResources/MSFT_IntuneMobileAppsMacOSLobApp/MSFT_IntuneMobileAppsMacOSLobApp.psm1

```diff
@@ -0,0 +1,566 @@
+        $BodyParameter,
+
+        [Parameter()]
+        [PSObject[]]
+        $Assignments,
+
+        [Parameter()]
+        [PSObject[]]
+        $Categories,
+
+        [Parameter()]
+        [System.String]
+        $Description,
+
+        [Parameter()]
+        [System.String]
+        $Developer,
+
+        [Parameter()]
+        [System.String]
+        $DisplayName,
+
+        [Parameter()]
+        [System.String]
+        $InformationUrl,
+
+        [Parameter()]
+        [System.Management.Automation.SwitchParameter]
+        $IsFeatured,
+
+        [Parameter()]
+        [PSObject]
+        $LargeIcon,
+
+        [Parameter()]
+        [System.String]
+        $Notes,
+
+        [Parameter()]
+        [System.String]
+        $Owner,
+
+        [Parameter()]
+        [System.String]
+        $PrivacyInformationUrl,
+
+        [Parameter()]
+        [System.String]
+        $Publisher,
+
+        [Parameter()]
+        [PSObject]
+        $PublishingState,
+
+        [Parameter()]
+        [PSObject[]]
+        $Relationships,
+
+        #endregion
+
+        [Parameter()]
+        [ValidateSet('Present', 'Absent')]
+        [System.String]
+        $Ensure = 'Present',
+
+        [Parameter()]
+        [System.Management.Automation.PSCredential]
+        $Credential,
+
+        [Parameter()]
+        [System.String]
+        $ApplicationId,
+
+        [Parameter()]
+        [System.String]
+        $TenantId,
+
+        [Parameter()]
+        [System.String]
+        $CertificateThumbprint,
+
+        [Parameter()]
+        [Switch]
+        $ManagedIdentity,
+
+        [Parameter()]
+        [System.String[]]
+        $AccessTokens
+    )
+
+    New-M365DSCConnection -Workload 'MicrosoftGraph' `
+        -InboundParameters $PSBoundParameters | Out-Null
+
+    #Ensure the proper dependencies are installed in the current environment.
+    Confirm-M365DSCDependencies
+
+    #region Telemetry
+    $ResourceName = $MyInvocation.MyCommand.ModuleName.Replace('MSFT_', '')
+    $CommandName = $MyInvocation.MyCommand
+    $data = Format-M365DSCTelemetryParameters -ResourceName $ResourceName `
+        -CommandName $CommandName `
+        -Parameters $PSBoundParameters
+    Add-M365DSCTelemetryEvent -Data $data
+    #endregion
+
+    $nullResult = $PSBoundParameters
+    $nullResult.Ensure = 'Absent'
+    try
+    {
+        if ($null -ne $Script:exportedInstances -and $Script:ExportMode)
+        {
+            $instance = $Script:exportedInstances | Where-Object -FilterScript {$_.Id -eq $Id}
+        }
+        else
+        {
+            $instance = Get-MgDeviceAppManagementMobileApp -MobileAppId $Id -ErrorAction Stop
+        }
+
+        if ($null -eq $instance)
+        {
+            Write-Verbose -Message "No Mobile app with Id {$Id} was found. Search with DisplayName."
+            $policy = Get-MgBetaDeviceAppManagementiOSManagedAppProtection -Filter "displayName eq '$DisplayName'" -ErrorAction SilentlyContinue
```

Comment:
```suggestion
            $policy = Get-MgBetaDeviceAppManagementMobileApp -Filter "displayName eq '$DisplayName'" -ErrorAction SilentlyContinue
```

Note: The `LargeIcon` property is not fetched when searching by display name. It is only available if the call is done against a specific mobile app id.

Reply:
Oh, thanks for pointing out that LargeIcon is fetched only with Id. Then should I even bother to include it in this PR? We are not supposed to use the 'Id' to be tenant agnostic, right?

Reply:
@Kajalp1079 It should be included. Please don't misunderstand it when I say "only available when fetched with Id". That doesn't mean that you have to make the Id a key parameter or something like that, it simply means that the contents of `LargeIcon` are only available to you when you call `Get-MgBetaDeviceAppManagementMobileApp -MobileAppId <XYZ>`. If you don't call it specifically like this, e.g. with `-Filter "displayName eq '<XYZ'"`, then `LargeIcon` will be an empty object. 

That means the following: If there is no policy with the provided Id, look it up by display name. If there is a hit on it, then call the policy directly to retrieve the contents of `LargeIcon`, and continue with the execution flow.